Terraform v1.5.6, terraform-provider-azurerm_v3.71.0_x5 is crashing while running plan or apply

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

v1.5.6

### AzureRM Provider Version

v3.71.0

### Affected Resource(s)/Data Source(s)

azurerm_api_management_backend

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

terraform {
  backend "azurerm" {
  ...
  }
}
```


### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.71.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104637490]

goroutine 14933 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement.flattenApiManagementBackendCredentials(0x140049ae540)
	github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/api_management_backend_resource.go:544 +0x330
github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement.resourceApiManagementBackendRead(0x140020ad500, {0x10674d0c0?, 0x140027f8480?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/api_management_backend_resource.go:358 +0x4c0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x10753c420?, {0x10753c420?, 0x14001b1c990?}, 0xd?, {0x10674d0c0?, 0x140027f8480?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:712 +0x134
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x14000b0a0e0, {0x10753c420, 0x14001b1c990}, 0x140008e5c70, {0x10674d0c0, 0x140027f8480})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:1015 +0x46c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x1400000d8a8, {0x10753c420?, 0x14001b1c870?}, 0x1400366c680)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:613 +0x400
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x14001541860, {0x10753c420?, 0x14001ed1dd0?}, 0x1400314d620)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:748 +0x3e8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x107136100?, 0x14001541860}, {0x10753c420, 0x14001ed1dd0}, 0x140011a4380, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001c21e0, {0x10754c9c0, 0x140018349c0}, 0x14003ba7440, 0x140017b3170, 0x10be37cb0, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1336 +0xbac
google.golang.org/grpc.(*Server).handleStream(0x140001c21e0, {0x10754c9c0, 0x140018349c0}, 0x14003ba7440, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1704 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.53.0/server.go:965 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.53.0/server.go:963 +0x28c

Error: The terraform-provider-azurerm_v3.71.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```
```


### Expected Behaviour

successful apply

### Actual Behaviour

provider crashed

there more than 800 other resources also in the same root module, not sure if it causing because of number of resources

### Steps to Reproduce

terraform plan
or 
terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/apimanagement/api_management_backend_resource.go
+++ b/internal/services/apimanagement/api_management_backend_resource.go
@@ -540,17 +540,17 @@ func flattenApiManagementBackendCredentials(input *backend.BackendCredentialsCon
 	if input.Certificate != nil {
 		result["certificate"] = *input.Certificate
 	}
-	result["header"] = flattenApiManagementBackendCredentialsObject(*input.Header)
-	result["query"] = flattenApiManagementBackendCredentialsObject(*input.Query)
+	result["header"] = flattenApiManagementBackendCredentialsObject(input.Header)
+	result["query"] = flattenApiManagementBackendCredentialsObject(input.Query)
 	return append(results, result)
 }
 
-func flattenApiManagementBackendCredentialsObject(input map[string][]string) map[string]interface{} {
+func flattenApiManagementBackendCredentialsObject(input *map[string][]string) map[string]interface{} {
 	results := make(map[string]interface{})
-	if input == nil {
+	if input == nil || *input == nil {
 		return results
 	}
-	for k, v := range input {
+	for k, v := range *input {
 		results[k] = strings.Join(v, ",")
 	}
 	return results

--- a/internal/services/apimanagement/api_management_backend_resource.go
+++ b/internal/services/apimanagement/api_management_backend_resource.go
@@ -547,7 +547,7 @@ func flattenApiManagementBackendCredentials(input *backend.BackendCredentialsCon
 
 func flattenApiManagementBackendCredentialsObject(input *map[string][]string) map[string]interface{} {
 	results := make(map[string]interface{})
-	if input == nil || *input == nil {
+	if input == nil {
 		return results
 	}
 	for k, v := range *input {

